vectorstores/mongovector/mongovector_test.go

```diff
@@ -0,0 +1,592 @@
+package mongovector
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/tmc/langchaingo/embeddings"
+	"github.com/tmc/langchaingo/schema"
+	"github.com/tmc/langchaingo/vectorstores"
+	"go.mongodb.org/mongo-driver/v2/bson"
+	"go.mongodb.org/mongo-driver/v2/mongo"
+	"go.mongodb.org/mongo-driver/v2/mongo/options"
+)
+
+// Run the test without setting up the test space.
+//
+//nolint:gochecknoglobals
+var testWithoutSetup = flag.Bool("no-atlas-setup", false, "don't create required indexes")
+
+const (
+	testDB                    = "langchaingo-test"
+	testColl                  = "vstore"
+	testIndexDP1536           = "vector_index_dotProduct_1536"
+	testIndexDP1536WithFilter = "vector_index_dotProduct_1536_w_filters"
+	testIndexDP3              = "vector_index_dotProduct_3"
+	testIndexSize1536         = 1536
+	testIndexSize3            = 3
+)
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+
+	defer func() {
+		os.Exit(m.Run())
```

Comment:
this is pretty odd IMO, can we skip this defer and just wrap the setup in a conditional vs the early return?